clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
Not sure how using Collection would really help -- the element type for Maps would be `Entry<TopicPartition, OffsetAndMetadata>`, which won't implement any useful interface that would be interchangeable with a `List<OffsetCommitInfo>` where the `OffsetCommitInfo` contains both the topic partition and the offset and metadata.

For mutability, in Copycat I just handled this at the application level -- subsequent updates are handled in a secondary map and they are merged once the commit completes. Would be nice to avoid having to do this if possible.